src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs

```diff
@@ -91,10 +91,7 @@ public class OpenIdConnectMiddleware : AuthenticationMiddleware<OpenIdConnectOpt
                 throw new ArgumentException("Options.SignInScheme is required.");
             }
 
-            if (Options.HtmlEncoder == null)
-            {
-                Options.HtmlEncoder = services.GetHtmlEncoder();
-            }
+            HtmlEncoder = services.GetHtmlEncoder();
```

Comment:
If HtmlEncoder it's not coming from options anymore then you can directly inject it via the constructor params like IUrlEncoder.